core/plugins/org.polarsys.capella.core.compare/src/org/polarsys/capella/core/compare/CapellaScope.java

```diff
@@ -282,6 +292,36 @@ protected IFile getFileFor(Resource resource) {
     return result;
   }
   
+  /**
+   * Return a scope-local element that corresponds to the given one, if any, otherwise return
+   * the given element
+   * @param element a potentially null element
+   * @return an element that is null if and only if the given element is null
+   */
+  protected EObject getLocalElement(EObject element) {
+    EObject result = element;
+    ResourceSet localResourceSet = _resourceSet;
+    if (element != null && localResourceSet != null ) {
+      Resource valueResource = element.eResource();
+      if (valueResource != null && localResourceSet != valueResource.getResourceSet()) {
+        // Element belongs to another resource set
+        URI valueResourceURI = valueResource.getURI();
+        if (valueResourceURI != null && valueResourceURI.isPlatformPlugin()) {
+          // And element belongs to a platform plug-in
+          URI valueURI = EcoreUtil.getURI(element);
+          if (valueURI != null) {
+            // Try and get the same element (URI-based) in the local resource set
+            EObject localElement = localResourceSet.getEObject(valueURI, false);
```

Comment:
Are you sure about the `false`?
Does external resources are loaded as long as scope is used in diffmerge?

Reply:
The rationale is to have as little impact as possible: if the desired element is not readily available, then the usual behavior is executed. It does not otherwise seem harmful to me because we are in a context (scope modification through add(...)) where the scope is supposed to have been loaded already. Also, the `FragmentedModelScope` base class was not intended to allow additional loading (using `true`) from a modification method such as `add(...)`.

Reply:
Ok, i have updated junit accordingly